roachtest: disable 4K block size intent resolution test

andrewbaptist opened the pull request:
With a 4K block size, the intent resolution test will cause unacceptable slowdowns. This commit changes the perturbation/*/intents to only test up to 1024 block sizes.

This also reduces the max perturbation duration to 10 minutes.

Informs: #139187
Informs: #139188
Informs: #137590 
Informs: #135934
Fixes: #137590

Epic: none

Release note: None

Commit:
roachtest: disable 4K block size intent resolution test

With a 4K block size, the intent resolution test will cause unacceptable
slowdowns. This commit changes the perturbation/*/intents to only test
up to 1024 block sizes.

Informs: #139187
Informs: #139188
Informs: #137590

Epic: none

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/139192)
<!-- Reviewable:end -->

Commit:
roachtest: disable perturbation mode with too many intents

Previously the perturbation/metamorphic/intents test could fail if there
were too many intents created. This change reduces the duration of
intent creation to a maximum of 10 minutes to prevent this failure mode.

Informs: #135934
Fixes: #137590

Release note: None

pav-kv commented on pkg/cmd/roachtest/tests/perturbation/intents.go:
```diff
@@ -33,7 +33,22 @@ func (i intents) setup() variations {
+	// are written.
+	// TODO(139188): Large block sizes result in a big slowdown when the intents
+	// are resolved.
+	if v.blockSize > 1024 {
```
optional: `v.blockSize = min(v.blockSize, 1024)` for brevity
Same below.

andrewbaptist replied:
Good idea - I'm going to switch to this style for all of these as it is probably easier to read.

pav-kv commented on pkg/cmd/roachtest/tests/perturbation/intents.go:
```diff
@@ -33,7 +33,22 @@ func (i intents) setup() variations {
+	v := i.setup()
+	v = v.randomize(rng)
+
+	// TODO(139187): Large block sizes result in a big slowdown when the intents
```
nit: `TODO(#139187)`
Passim.

andrewbaptist commented:
TFTR!

bors r=pav-kv
